src/components/toolbar/menu-bar/view-menu.tsx

```diff
@@ -0,0 +1,86 @@
+import i18n from '@dhis2/d2-i18n'
+import type { FC } from 'react'
+import { useCallback } from 'react'
+import { ACCESSORY_PANEL_DEFAULT_WIDTH } from '@constants/panels'
+import {
+    HoverMenuDropdown,
+    HoverMenuList,
+    HoverMenuListItem,
+} from '@dhis2/analytics'
+import { useAppDispatch, useAppSelector } from '@hooks'
+import { setUserSidebarWidthToLocalStorage } from '@modules/local-storage'
+import { getCurrentId } from '@store/current-slice'
+import {
+    getUiAccessoryPanelWidth,
+    getUiDetailsPanelOpen,
+    getUiLayoutPanelHidden,
+    getUiSidebarHidden,
+    toggleUiLayoutPanelHidden,
+    toggleUiSidebarHidden,
+    setUiAccessoryPanelWidth,
+    setUiDetailsPanelOpen,
+} from '@store/ui-slice'
+
+export const ViewMenu: FC = () => {
+    const dispatch = useAppDispatch()
+
+    const isSidebarHidden = useAppSelector(getUiSidebarHidden)
+    const isLayoutPanelHidden = useAppSelector(getUiLayoutPanelHidden)
+    const isDetailsPanelOpen = useAppSelector(getUiDetailsPanelOpen)
+    const userSettingWidth = useAppSelector(getUiAccessoryPanelWidth)
+    const id = useAppSelector(getCurrentId)
+
+    const toggleLayoutPanelHidden = useCallback(() => {
```

Comment:
I guess in these cases we don't need to use `useCallback`, what do you think?

```suggestion
    const toggleLayoutPanelHidden = () => dispatch(toggleLayoutPanelHidden())
```

Reply:
This is a topic for heated debates. `dispatch` is a stable reference, so the `toggleLayoutPanelHidden` version that does `useCallback` and only has `dispatch` in its dependency array is also a stable reference. I think this is nice, because now if you pass it to child components, use it in a `useEffect` hook, or whatever, it will never cause unexpected rerenders.

On the other hand, others say it is an anti-pattern. And could also make things slower. So you should always use inline functions that get recreated at every render cycle (like your suggestion), unless that causes issues (either rerendering or performance).

I personally tend to do pretty much the opposite these days. I basically use `useCallback` pretty much everywhere. The only exception is when I write JSX and I attach event listeners to built-in element (i.e.`input`, `button`, etc.), because for these I know for a fact that it won't ever cause issues.

In my personal experience, the most common type of issue in React is that components rerender or effects run when you don't want/expect them to. So I am always trying to keep everything "as stable as possible". This means I always use `useMemo` and `useCallback`. It may be a bit controversial though.

Reply:
Ok, I saw you used `useCallback` in other parts of the app and I did the same, but then realised in my case there are no dependencies so I had a doubt.
I'm fine with keeping it and have the same approach everywhere.